misk-zookeeper/src/test/kotlin/misk/clustering/zookeeper/ZkTestModule.kt

```diff
@@ -29,20 +29,32 @@ internal class ZkTestModule : KAbstractModule() {
     install(ZookeeperModule())
   }
 
-  @Singleton
-  class StartZookeeperService : AbstractIdleService(), DependentService {
+  @Singleton class StartZookeeperService : AbstractIdleService(), DependentService {
     override val consumedKeys: Set<Key<*>> = setOf()
     override val producedKeys: Set<Key<*>> = setOf(startZkServiceKey)
-    private val sharedZookeeper = EmbeddedZookeeper(zkPortKey)
 
     override fun startUp() {
-      log.info { "starting zk instance " }
-      sharedZookeeper.start()
+      if (!hasStarted) {
```

Comment:
Depending on how aggressively gradle is parallelizing, you may want to make this an `AtomicBoolean`?